services/src/main/java/org/keycloak/authentication/authenticators/client/FederatedJWTClientAuthenticator.java

```diff
@@ -139,4 +146,22 @@ public boolean isSupported(Config.Scope config) {
         return Profile.isFeatureEnabled(Profile.Feature.CLIENT_AUTH_FEDERATED);
     }
 
+    protected static String toIssuer(String subject) {
+        try {
+            URI uri = new URI(subject);
+            String scheme = uri.getScheme();
+            String rawAuthority = uri.getRawAuthority();
+            if (scheme == null || rawAuthority == null) {
+                return null;
+            }
+            StringBuilder sb = new StringBuilder();
```

Comment:
No need or the StringBuilder, we can just do `return scheme + "://" + rawAuthority;`